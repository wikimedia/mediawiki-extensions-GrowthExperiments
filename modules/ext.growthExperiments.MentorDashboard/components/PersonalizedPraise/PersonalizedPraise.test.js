jest.mock( '../../../vue-components/icons.json', () => ( {
	cdxIconNext: '',
	cdxIconPrevious: '',
	cdxIconSettings: '',
	cdxIconUserAvatar: ''
} ), { virtual: true } );
const { mount, shallowMount } = require( '@vue/test-utils' );
const Vuex = require( 'vuex' );
const PersonalizedPraise = require( './PersonalizedPraise.vue' );
const praiseworthyMenteeJSON = require( '../../../../tests/qunit/__mocks__/praiseworthyMentee.json' );

describe( 'PersonalizedPraise', () => {
	let praiseworthyMenteesActions, praiseworthyMenteesGetters;
	let hasData = false;
	let store;

	beforeEach( () => {
		praiseworthyMenteesActions = {
			fetchMentees: jest.fn( () => $.Deferred().resolve().promise() ),
			previousPage: jest.fn( () => $.Deferred().resolve().promise() ),
			nextPage: jest.fn( () => $.Deferred().resolve().promise() ),
			saveSettings: jest.fn( () => $.Deferred().resolve().promise() )
		};
		praiseworthyMenteesGetters = {
			totalPages: jest.fn( () => {
				if ( hasData ) {
					return 1;
				} else {
					return 0;
				}
			} ),
			currentPage: jest.fn( () => {
				if ( hasData ) {
					return 1;
				} else {
					return 0;
				}
			} ),
			mentee: jest.fn( () => {
				if ( hasData ) {
					return praiseworthyMenteeJSON;
				} else {
					return undefined;
				}
			} ),
			settings: jest.fn( () => ( {
				minEdits: 1,
				days: 14,
				messageSubject: 'Message subject',
				messageText: 'Message text',
				notificationFrequency: 0
			} ) )
		};
		store = new Vuex.Store( {
			modules: {
				praiseworthyMentees: {
					actions: praiseworthyMenteesActions,
					getters: praiseworthyMenteesGetters,
					namespaced: true
				}
			}
		} );
	} );

	it( 'NoResult when no mentees', () => {
		hasData = false;
		const wrapper = mount( PersonalizedPraise, {
			global: {
				provide: {
					$log: jest.fn()
				},
				mocks: {
					log: jest.fn(),
					$store: store,
					$filters: {
						convertNumber: jest.fn( ( x ) => `${x}` )
					}
				}
			}
		} );
		expect( wrapper.element ).toMatchSnapshot();
	} );

	it( 'full data when has mentees', () => {
		hasData = true;
		const wrapper = shallowMount( PersonalizedPraise, {
			global: {
				provide: {
					$log: jest.fn()
				},
				mocks: {
					log: jest.fn(),
					$store: store,
					$filters: {
						convertNumber: jest.fn( ( x ) => `${x}` )
					}
				}
			}
		} );
		expect( wrapper.element ).toMatchSnapshot();
	} );
} );
