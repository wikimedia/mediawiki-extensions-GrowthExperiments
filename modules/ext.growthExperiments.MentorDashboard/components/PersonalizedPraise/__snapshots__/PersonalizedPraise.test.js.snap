// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PersonalizedPraise NoResult when no mentees 1`] = `
<section
  class="ext-growthExperiments-PersonalizedPraise"
>
  <p>
    growthexperiments-mentor-dashboard-personalized-praise-metrics (1,14)
  </p>
  <div
    class="ext-growthExperiments-PersonalizedPraise-Settings"
  >
    <button
      aria-label="growthexperiments-mentor-dashboard-personalized-praise-settings-cog-icon-label"
      class="cdx-button cdx-button--action-default cdx-button--weight-quiet cdx-button--icon-only ext-growthExperiments-PersonalizedPraise-Settings__cogicon"
    >
      
      <span
        class="cdx-icon cdx-icon--medium"
      >
        <svg
          aria-hidden="true"
          height="20"
          viewBox="0 0 20 20"
          width="20"
          xmlns="http://www.w3.org/2000/svg"
        >
          <!---->
          <path
            d=""
          />
        </svg>
      </span>
      
    </button>
    <transition-stub>
      <!---->
    </transition-stub>
  </div>
  <div
    class="ext-growthExperiments-NoResults"
  >
    <div
      class="ext-growthExperiments-NoResults__image"
    />
    <p>
      {}
    </p>
  </div>
  <!--v-if-->
  <!--v-if-->
</section>
`;

exports[`PersonalizedPraise full data when has mentees 1`] = `
<section
  class="ext-growthExperiments-PersonalizedPraise"
>
  <p>
    growthexperiments-mentor-dashboard-personalized-praise-metrics (1,14)
  </p>
  <personalized-praise-settings-stub
    settingsdata="[object Object]"
  />
  <!--v-if-->
  <user-card-stub
    mentee="[object Object]"
  />
  <personalized-praise-pagination-stub
    currentpage="1"
    totalpages="1"
  />
</section>
`;
